flintrock/services.py

```diff
@@ -348,7 +348,26 @@ def configure(
             'spark/conf/slaves',
             'spark/conf/spark-defaults.conf',
         ]
+
+        # Export SPARK_DIST_CLASSPATH if user wants to use Spark's "Hadoop Free" Build.
+        # Cf https://spark.apache.org/docs/2.2.0/hadoop-provided.html
+        # This allows us to use Hadoop 2.8.
+        # NB: slf4jars are excluded, to avoid error "SLF4J: Class path contains multiple SLF4J bindings" (when
+        # multiple bindings are found, Spark logs are not written)
+        spark_dist_classpath = \
```

Comment:
Using `(...)` is better for long one-liners. `\` is ugly. :)
For sed, the first character after `s` is the separator, so you can use `;` instead of `/` to avoid escaping all the `/` in the path.
I think this should work (not tested):
```
spark_dist_classpath = (
    'export SPARK_DIST_CLASSPATH=$(hadoop classpath | '
    'sed "s;/home/ec2-user/hadoop/share/hadoop/common/lib/slf4j[-\.a-z0-9]*\.jar:;;g")'
)
    if 'without-hadoop' in self.download_source else ''
```